core/arch/arm/plat-imx/conf.mk

```diff
@@ -83,12 +89,33 @@ CFG_BOOT_SECONDARY_REQUEST ?= y
 CFG_ENABLE_SCTLR_RR ?= y
 endif
 
+# i.MX6 Solo/DualLite/Dual/Quad specific config
+ifeq ($(filter y, $(CFG_MX6Q) $(CFG_MX6D) $(CFG_MX6DL) $(CFG_MX6S)), y)
+CFG_TZDRAM_START ?= 0x4E000000
+CFG_TZDRAM_SIZE ?= 0x01F00000
+CFG_SHMEM_START ?= ($(CFG_TZDRAM_START) + $(CFG_TZDRAM_SIZE))
+CFG_SHMEM_SIZE ?= 0x00100000
+endif
+
+# i.MX6 SoloX specific config
+ifeq ($(filter y, $(CFG_MX6SX)), y)
+CFG_TZDRAM_START ?= (0x80000000 + $(CFG_DDR_SIZE) - 0x02000000)
```

Comment:
This needs to be `0x80000000 - 0x02000000 + $(CFG_DDR_SIZE)`, CFG_DDR_SIZE sometimes is 2GB, so first subtract the shmem size will avoid errors.

Reply:
I met compile issue before. Seems no issue now.

Reply:
Might depends on compilation toolchain.
I would recommend `CFG_TZDRAM_START ?= (0x80000000 - 0x02000000 + $(CFG_DDR_SIZE))`, but i can't tell its bulletproof.